Include/cpython/unicodeobject.h

```diff
@@ -146,7 +146,7 @@ typedef struct {
         unsigned int statically_allocated:1;
         /* Padding to ensure that PyUnicode_DATA() is always aligned to
            4 bytes (see issue #19537 on m68k). */
-        unsigned int :24;
+        unsigned int :10;
```

Comment:
IIRC, the bit size of `unsigned int` is greater or equal to 16 (most of the size is 32 bits, but some systems use 16 bits).
I maintain unused size as 10-bit. But tackle me if I am wrong.

Reply:
This should be `18` after the `interned` field is moved back into `state`:

* interned = 8 bits
* kind = 3 bits
* compact = 1 bit
* ascii = 1 bit
* statically_allocated = 1 bit
* padding = 18 bits

For a total of 32 bits (4 bytes).

We don't support any platforms with 16-bit ints. This might fail to compile on those platforms, but that was already true with the existing `unsigned int :24` and I don't think it's worth changing it. (For example, AVR GCC fails with "error: width of '<anonymous>' exceeds its type")